File: app/src/main/java/knf/kuma/tv/BindableCardView.java
package knf.kuma.tv;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.ImageView;

import androidx.annotation.LayoutRes;
import androidx.leanback.widget.BaseCardView;

public abstract class BindableCardView<T> extends BaseCardView {

    public Context context;

    public BindableCardView(Context context) {
        super(context);
        this.context = context;
        initLayout();
    }

    public BindableCardView(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.context = context;
        initLayout();
    }

    public BindableCardView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        this.context = context;
        initLayout();
    }

    private void initLayout() {
        setFocusable(true);
        setFocusableInTouchMode(true);
        LayoutInflater inflater = LayoutInflater.from(getContext());
        inflater.inflate(getLayoutResource(), this);
    }

    public abstract void bind(T data);

    public abstract ImageView getImageView();

    public abstract @LayoutRes
    int getLayoutResource();
}

File: app/src/main/java/knf/kuma/tv/BindableCardView.kt
package knf.kuma.tv

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.ImageView

import androidx.annotation.LayoutRes
import androidx.leanback.widget.BaseCardView

abstract class BindableCardView<T> : BaseCardView {

    abstract val imageView: ImageView

    @get:LayoutRes
    abstract val layoutResource: Int

    constructor(context: Context) : super(context) {
        initLayout()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        initLayout()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        initLayout()
    }

    private fun initLayout() {
        isFocusable = true
        isFocusableInTouchMode = true
        val inflater = LayoutInflater.from(context)
        inflater.inflate(layoutResource, this)
    }

    abstract fun bind(data: T)
}
